src/devices/Board/AlternatePinMode.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Iot.Device.Board
+{
+    /// <summary>
+    /// Alternate pin modes, for pins that support them
+    /// </summary>
+    public enum AlternatePinMode
+    {
+        // Note: Do not change the enum values without reviewing the impact
+        Unknown = -2,
+        NotSupported = -1,
+        Gpio = 0,
+
+        // the names are Raspi specific, but this is considered a generic representation of alternate modes (if any)
+        Alt0 = 1,
```

Comment:
I'm wondering if this should rather be: Pwm0, Pwm1, Spi0Mosi, etc (hope you got the idea), the Alt0, Alt1... seem very generic, might be fine as an intermediate abstraction layer but it's not very helpful unless you got a spec in front of you

Reply:
The user on the top-level API will not have to worry what these mean. Even though `GetHardwareModeForPinUsage` is public, the user rarely needs to use this directly.